File: CPP/118.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {

        vector<vector<int>> output;
        if (!numRows) return output;

        // First row
        output.push_back(vector<int> {1});
        // Creating each row
        for (auto i = 1; i < numRows; ++i){
            vector<int> row;
            // Each row elements
            for (auto j = 0; j <= i; ++j){
                // first and last element = 1
                if (j == 0 || j == i ) row.push_back(1);
                else {
                    // Elements in between
                    // if j -> add (output[i-1][j-1][j])
                    row.push_back(output[i-1][j-1] + output[i-1][j]);
                }
            }
            output.push_back(row);
        }

        return output;
    }
};

File: Python/118.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        # List[List[int]] == [[]] ....?
        output = []
        if not numRows: return output
        output.append([1])

        for rowInd in range(1, numRows):
            row = []
            for elemInd in range(0, rowInd+1):
                if elemInd == 0 or elemInd == rowInd:
                    row.append(1)
                else:
                    row.append(output[rowInd-1][elemInd-1] + output[rowInd-1][elemInd])
            output.append(row)

        return output
